app/services/OdsService.scala

```diff
@@ -18,36 +18,87 @@ package services
 
 import cats.data.EitherT
 import com.google.inject.Inject
-import connectors.ODSConnector
+import connectors.SelfAssessmentODSConnector
 import models._
+import play.api.http.Status.NOT_FOUND
 import play.api.libs.json.{JsValue, Json}
+import play.api.mvc.Request
 import uk.gov.hmrc.http.{HeaderCarrier, UpstreamErrorResponse}
 import utils.TaxsJsonHelper
 
 import scala.concurrent.{ExecutionContext, Future}
 
 class OdsService @Inject() (
   jsonHelper: TaxsJsonHelper,
-  odsConnector: ODSConnector
+  selfAssessmentOdsConnector: SelfAssessmentODSConnector
 )(implicit ec: ExecutionContext) {
 
-  def getPayload(UTR: String, TAX_YEAR: Int)(implicit
-    hc: HeaderCarrier
+  def getPayload(utr: String, TAX_YEAR: Int)(implicit
+    hc: HeaderCarrier,
+    request: Request[_]
   ): EitherT[Future, UpstreamErrorResponse, JsValue] =
     for {
-      taxpayer     <- odsConnector.connectToSATaxpayerDetails(UTR).map(_.json.as[JsValue])
-      taxSummaries <- odsConnector.connectToSelfAssessment(UTR, TAX_YEAR).map(_.json.as[JsValue])
-    } yield jsonHelper.getAllATSData(taxpayer, taxSummaries, UTR, TAX_YEAR)
+      taxpayer     <- selfAssessmentOdsConnector
+                        .connectToSATaxpayerDetails(utr)
+                        .transform {
+                          case Right(response) if response.status == NOT_FOUND =>
+                            Left(UpstreamErrorResponse("NOT_FOUND", NOT_FOUND))
+                          case Right(response)                                 => Right(response.json.as[JsValue])
+                          case Left(error)                                     => Left(error)
+                        }
+      taxSummaries <- selfAssessmentOdsConnector.connectToSelfAssessment(utr, TAX_YEAR).transform {
+                        case Right(response) if response.status == NOT_FOUND =>
```

Comment:
Noticed you also changed this in the tests
Why is NOT_FOUND in a Right rather than a Left now?

Reply:
Because of the cache. The cache is caching an http response. By pushing the not found into a http response so a right it gets cached automatically. The service then turn it back into a left so it feeds into the existing logic